tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -44,7 +44,7 @@
 
 # How much memory to assign to redpanda per partition. Redpanda will be started
 # with MIB_PER_PARTITION * PARTITIONS_PER_SHARD * CORE_COUNT memory
-DEFAULT_MIB_PER_PARTITION = 4
+DEFAULT_MIB_PER_PARTITION = 3.3
```

Comment:
This magic number could use a comment, describing what is going on here (or a pointer to a JIRA which does same).

Reply:
I thought the existing comment needed less explanation since it was 4 MB which was exactly the default value for the "mem per partition" config (before your changes), but I see this is actually referring to how we calculate --memory for this test.

Reply:
I am confused by this property: the comment indicates that it is used to calculate the `--memory` argument. Is that right? But it _also_ used to set `topic_memory_per_partition` (I had thought it was only used to set this latter prop).

If it's used to set both it seems like changing this would be a no-op wrt the partition memory limit (e.g., doubling would double the `--memory` and partition memory bucket, but also double the memory per part, so cancel out)?

Reply:
Keep in mind this is on 24.3 so doesn't have any of the newer changes.

> the comment indicates that it is used to calculate the --memory argument. Is that right? But it also used to set topic_memory_per_partition (I had thought it was only used to set this latter prop).

Yes it's both

> If it's used to set both it seems like changing this would be a no-op wrt the partition memory limit (e.g., doubling would double the --memory and partition memory bucket, but also double the memory per part, so cancel out)?

Well yes but the problem is that if it calculates `--memory` to 6GB per shard you won't be able to start on a 4GB per shard machine which is what we are running on now. Hence we need to lower it.

Reply:
OK, so we are actually testing something a bit more dense than the nominal limit, i.e., we document 1000/4MB, but we actually test 1000/3.3MB?

SGTM.

Reply:
Correct yes